crates/delegator/src/main.rs

```diff
@@ -51,71 +40,36 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     // Generate topic
     let new_job_topic = gossipsub_ident_topic(network, Topic::NewJob);
     let picked_job_topic = gossipsub_ident_topic(network, Topic::PickedJob);
+    let finished_job_topic = gossipsub_ident_topic(network, Topic::FinishedJob);
 
-    let mut swarm_runner = SwarmRunner::new(
-        node_account.get_keypair(),
-        &[new_job_topic.to_owned(), picked_job_topic],
-    )?;
-
-    let (send_topic_tx, send_topic_rx) = mpsc::channel::<Vec<u8>>(1000);
-    let mut message_stream = swarm_runner.run(new_job_topic, send_topic_rx);
-    let mut event_stream = registry_handler.subscribe_events(vec!["0x0".to_string()]);
+    let (swarm_events_tx, swarm_events_rx) = mpsc::channel::<gossipsub::Event>(100);
+    let (job_witness_tx, job_witness_rx) = broadcast::channel::<JobWitness>(100);
 
-    let compiler = CairoCompiler::new(node_account.get_signing_key(), registry_address);
+    let (new_job_topic_tx, new_job_topic_rx) = mpsc::channel::<Vec<u8>>(100);
 
-    let mut compiler_scheduler =
-        FuturesUnordered::<Process<'_, Result<Job, CompilerControllerError>>>::new();
+    SwarmRunner::new(
+        node_account.get_keypair(),
+        vec![new_job_topic.to_owned(), picked_job_topic, finished_job_topic],
+        vec![(new_job_topic.to_owned(), new_job_topic_rx)],
+        swarm_events_tx,
+    )?;
 
-    // Read cairo program path from stdin
-    let mut stdin = BufReader::new(stdin()).lines();
+    Delegator::new(job_witness_tx, swarm_events_rx);
```

Comment:
I think we should view this `crates/delegator` binary as whole "Delegator". 
So imo the object name is more accurate if we change into `BehaviorHandler`.
```suggestion
    BehaviorHandler::new(job_witness_tx, swarm_events_rx);
```